src/reverse-proxy-handler.ts

```diff
@@ -91,6 +91,8 @@ export class ReverseProxyHandler implements ProxyHandler<ReverseProxyTarget> {
         setPrototypeOf(shadowTarget, env.getRawValue(secureProto));
         // defining own descriptors
         copyReverseOwnDescriptors(env, shadowTarget, target);
+        // reserve proxies are always frozen
+        freeze(shadowTarget);
```

Comment:
👆 will that freeze the shadow target object which is later proxied
(is `shadowTarget` the raw shadow object record or a proxy of the raw shadow object record)?

Reply:
maybe that's too much! but until now... I haven't found any DOM API that expects to mutate an object that was passed into it, so this protection seems to be fine. additionally, if we think about postMessage communication, you will never be able to mutate anything that is coming thru because it doesn't really change on the other end.

Reply:
Ah, I see! `shadowTarget` is created via `createReverseShadowTarget()`. So there is the real object, the object for reverse proxies, and the object for secure realm. Also unlike secure and raw objects which are stored in weakmaps the reverse proxied targets are created and frozen fresh for each encounter.